packages/preact/src/search/resultCaching.ts

```diff
@@ -1,56 +1,71 @@
-import type { SearchQuery, SearchResult } from "@nosto/nosto-js/client"
+import { SearchQuery, SearchResult } from "@nosto/nosto-js/client"
 import { isEqual } from "@utils/isEqual"
+import { isPlainObject } from "@utils/isPlainObject"
 import { getSessionStorageItem, setSessionStorageItem } from "@utils/storage"
 
 export const STORAGE_ENTRY_NAME = "nosto:search:searchResult"
 
-type SearchResultDto = {
+const pageQueryFields = ["size", "from"]
+
+export type SearchResultDto = {
   query: ReturnType<typeof getCacheKey>
   result: SearchResult
 }
 
-export function cacheSearchResult(usePersistentCache: boolean, query: SearchQuery, result: SearchResult) {
-  if (!usePersistentCache) {
-    return
-  }
-
-  const dto: SearchResultDto = { query, result }
-  setSessionStorageItem(STORAGE_ENTRY_NAME, dto)
-}
-
-export function loadCachedResultIfApplicable(usePersistentCache: boolean, query: SearchQuery): SearchResult | null {
+export function loadCachedResultIfApplicable(usePersistentCache: boolean, query: SearchQuery) {
   if (!usePersistentCache) {
     return null
   }
 
-  const storageValue = getSessionStorageItem(STORAGE_ENTRY_NAME)
-  if (!storageValue || !isValueShapeCorrect(storageValue)) {
+  const storageValue = getSessionStorageItem<SearchResultDto>(STORAGE_ENTRY_NAME)
+  if (!storageValue) {
     return null
   }
 
   const cachedQuery = getCacheKey(storageValue.query)
-  if (!isEqual(getCacheKey(query), cachedQuery)) {
+  if (!isEqual(getCacheKey(query), cachedQuery, pageQueryFields)) {
     return null
   }
-  return storageValue.result
+  return storageValue
 }
 
-function getCacheKey(query: SearchQuery): Omit<SearchQuery, "time"> {
-  return {
-    accountId: query.accountId,
-    customRules: query.customRules,
-    explain: query.explain,
-    keywords: query.keywords,
-    products: query.products,
-    query: query.query,
-    redirect: query.redirect,
-    rules: query.rules,
-    segments: query.segments,
-    sessionParams: query.sessionParams
+export function cacheSearchResult(usePersistentCache: boolean, query: SearchQuery, result: SearchResult) {
+  if (!usePersistentCache) {
+    return
   }
+
+  const dto: SearchResultDto = { query, result }
+  setSessionStorageItem(STORAGE_ENTRY_NAME, dto)
+}
+
+function getCacheKey<T extends object>(query: T): Omit<SearchQuery, "time"> {
```

Comment:
can you explain the benefit of this version compared to the previous version?

Reply:
The previous also included undefined data. It caused issues with comparisons. This version will have only the relevant details in the key

Reply:
isEqual doesn't handle undefined?

Reply:
No, it doesn't based on my testing

Reply:
then fix isEqual instead of adjusting getCacheKey